nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HealthCheckRebooter.java

```diff
@@ -0,0 +1,104 @@
+ *     enclosed by brackets [] replaced by your own identifying information:
+ *     "Portions Copyright [year] [name of copyright owner]"
+ *
+ *     Contributor(s):
+ *     If you wish your version of this file to be governed by only the CDDL or
+ *     only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *     elects to include this software in this distribution under the [CDDL or GPL
+ *     Version 2] license."  If you don't indicate a single choice of license, a
+ *     recipient has the option to distribute your version of this file under
+ *     either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *     its licensees as provided above.  However, if you add GPL Version 2 code
+ *     and therefore, elected the GPL Version 2 license, then the option applies
+ *     only if the new code is made subject to such option by the copyright
+ *     holder.
+ */
+package fish.payara.nucleus.healthcheck.admin;
+
+import com.sun.enterprise.util.SystemPropertyConstants;
+import fish.payara.nucleus.healthcheck.HealthCheckService;
+import fish.payara.nucleus.healthcheck.configuration.HealthCheckServiceConfiguration;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.I18n;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.*;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+
+import javax.inject.Inject;
+import java.util.Properties;
+
+/**
+ * Admin command to reboot health check service
+ *
+ * @author mertcaliskan
+ */
+@ExecuteOn({RuntimeType.DAS, RuntimeType.INSTANCE})
+@TargetType(value = {CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CLUSTERED_INSTANCE, CommandTarget.CONFIG})
+@Service(name = "reboot-healthcheck")
+@CommandLock(CommandLock.LockType.NONE)
+@PerLookup
+@I18n("reboot.healthcheck")
+@RestEndpoints({
+    @RestEndpoint(configBean = HealthCheckServiceConfiguration.class,
+            opType = RestEndpoint.OpType.POST,
+            path = "reboot-healthcheck",
+            description = "Reboots Health Check Service")
+})
+public class HealthCheckRebooter implements AdminCommand {
+
+    @Inject
+    ServerEnvironment server;
+
+    @Inject
+    HealthCheckService service;
+
+    @Inject
+    protected Target targetUtil;
+
+    @Param(name = "target", optional = true, defaultValue = SystemPropertyConstants.DAS_SERVER_NAME)
+    String target;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        final ActionReport actionReport = context.getActionReport();
+        Properties extraProperties = actionReport.getExtraProperties();
+        if (extraProperties == null) {
+            extraProperties = new Properties();
+            actionReport.setExtraProperties(extraProperties);
+        }
+
+        if (server.isDas()) {
```

Comment:
Could perhaps reduce complexity with:
```
if(server.isNotDas() || targetUtil.getConfig(target).isDas()){
	service.reboot();
}
```

Reply:
make sense, will apply it in a separate PR by going through all asadmin comands.